File: Dapper_pro/Models/Student.cs
using System.ComponentModel.DataAnnotations.Schema;
namespace Dapper_pro.Models
{
    public class Student
    
        {
        public int id { get; set; }
        public string username { get; set; }
        public string firstname { get; set; }
        public string lastname { get; set; }
        public string email { get; set; }
        public string statename {  get; set; }
        public int statecode {  get; set; }
        public string districtname { get; set; }
        public int districtcode {  get; set; }
        public int mobile { get; set; }
        public string status { get; set; }
        public string gender { get; set; }
    }



    public class searchmodel

    {
        public int pageIndex { get; set; }

        public int pageSize { get; set; }

        public string searchText { get; set; }


    }


    public class ProfileImageUpdateRequest
    {
        public string username { get; set; }
        public string profileImage { get; set; }
    }

    public class ProfileDetailUpdateRequest
    {
        public string username { get; set; }
        public string email { get; set; }
        public string profileImage { get; set; }
    }

    public class StdResetPasswordModel
    {
        public string Username { get; set; }
        public string OldPassword { get; set; }
        public string NewPassword { get; set; }
    }


    public class Attendance
    {
        public int Id { get; set; }
        public DateTime Date { get; set; }
        public string Status { get; set; }
        //public string Type { get; set; }
    }



    public class AttendanceStatus
    {
        public int StudentId { get; set; }
        public string Username { get; set; }
        public int DepartmentId { get; set; }
        public string DepartmentName { get; set; }
        public int Type { get; set; }
        public string Status { get; set; }
    }

    public class AttendanceSummary
    {
        public List<AttendanceStatus> AttendanceStatuses { get; set; }
        public int TotalPresent { get; set; }
        public int TotalAbsent { get; set; }
    }


    public class AttendanceResult
    {
        public DateTime Date { get; set; }
        public string Status { get; set; }
    }

    public class StudentSubjectMarks
    {
        public string subjectname { get; set; }
        public int marks { get; set; }
        public int A1marks {  get; set; }
        public int A2marks { get; set;}
        public int A3marks { get; set;}
    }


    public class StudentDetails
    {
        public int id { get; set; }
        public string firstname { get; set; }
        public string lastname { get; set; }
        public string departmentname { get; set; }
        public int departmentid { get; set; }
        public string subjectname { get; set; }
        public int subjectid { get; set; }
    }


}

File: Education_pro/src/app/types/student.ts
export interface Student{
  id:number;
  username:string;
  firstname:string;
  lastname:string;
  email:string;
  districtname:string;
  districtcode:number;
  statename:string;
  statecode:number;
  mobile:number;
  gender:string;
  
  departmentname: string;
  departmentid: number;
  subjectname: string | null;
  subjectid: number | null;

  marks?: number;
  a1marks?: number;
  a2marks?: number;
  a3marks?: number;
}

export interface DistrictCount {
    districtname: string;
    district_count: number;
  }


// +++++++++++++++++ using for chat  +++++++++++

  export interface ChatMessage {
    user: string;
    message: string;
    read: boolean;
  }




  export interface ResetPasswordModel {
    oldPassword: string;
    newPassword: string;
  }


  //+++++++++++++++++  using for the teacher ++++++++++++++

  export interface teacher{
    status: string;
    id:number;
    firstname:string;
    lastname:string;
    username:string;
    email:string;
    gender:string;
    districtname:string;
    districtcode:number;
    statename:string;
    statecode:number;
    mobile:number;
    qualifications:string;
  }


  // attendance-summary.model.ts

  export interface AttendanceStatus {
    teacherId: any;
    studentId:any;
    username:string;
    status: string;
    date: string;
    departmentName:string;
  }
  
  
  export interface attendanceTeacherStatuses {
    AttendanceStatuses: any;
    attendanceTeacherStatuses: AttendanceStatus[];
    totalPresent: number;
    totalAbsent: number;
  }
  export interface StudentAttendanceSummary {
    attendanceDetails: any;
    AttendanceStatuses: any;
    attendanceStatuses: AttendanceStatus[];
    totalPresent: number;
    totalAbsent: number;
  }

  export interface attendanceStatuses{
    AttendanceStatuses: any;
    attendanceStatuses:AttendanceStatus[];
    totalPresent:number;
    totalAbsent:number;
  }



  
export interface DetailedAttendance {
  teacherId: number;
  studentId:number;
  attendanceStatus: string;
  username:string;
  date:string
}


export interface TeacherCourseSubject {
  username:string;
  firstname: string;
  lastname: string;
  name: string;
  subjectname: string;
  assignment:string;
  departmentname:string;
  semester:number;
  marks: number;
}

export interface StudentSubjectMarks {
  firstname: string;
  departmentname: string;
  subjectname: string;
  marks: number;
  a1marks:number;
  a2marks:number;
  a3marks:number;
}






  




  
  
  
